packages/mui-base/src/PreviewCard/Popup/PreviewCardPopup.tsx

```diff
@@ -0,0 +1,81 @@
+'use client';
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { useComponentRenderer } from '../../utils/useComponentRenderer';
+import { PreviewCardPopupOwnerState, PreviewCardPopupProps } from './PreviewCardPopup.types';
+import { usePreviewCardRootContext } from '../Root/PreviewCardContext';
+import { usePreviewCardPositionerContext } from '../Positioner/PreviewCardPositionerContext';
+import { usePreviewCardPopup } from './usePreviewCardPopup';
+import { useForkRef } from '../../utils/useForkRef';
+
+const PreviewCardPopup = React.forwardRef(function PreviewCardPopup(
+  props: PreviewCardPopupProps,
+  forwardedRef: React.ForwardedRef<HTMLDivElement>,
+) {
+  const { className, render, ...otherProps } = props;
+
+  const { open, transitionStatus, getRootPopupProps, popupRef } = usePreviewCardRootContext();
+  const { side, alignment } = usePreviewCardPositionerContext();
+
+  const { getPopupProps } = usePreviewCardPopup({
+    getProps: getRootPopupProps,
+  });
+
+  const ownerState: PreviewCardPopupOwnerState = React.useMemo(
+    () => ({
+      open,
+      side,
+      alignment,
+      entering: transitionStatus === 'entering',
+      exiting: transitionStatus === 'exiting',
+    }),
+    [open, side, alignment, transitionStatus],
+  );
+
+  const mergedRef = useForkRef(popupRef, forwardedRef);
+
+  const { renderElement } = useComponentRenderer({
+    propGetter: getPopupProps,
+    ref: mergedRef,
+    render: render ?? 'div',
+    className,
+    ownerState,
+    extraProps: otherProps,
+    customStyleHookMapping: {
```

Comment:
This could be extracted to a const outside of the component so it's not recreated on every render

Reply:
I noticed our other components have it inlined sometimes, and `useComponentRenderer` passes it as a `memo` dependency, so they'll need to be updated as well